src/widget/systemtrayicon.cpp

```diff
@@ -116,6 +113,26 @@ QString SystemTrayIcon::extractIconToFile(QIcon icon, QString name)
     return iconPath;
 }
 
+#if defined(ENABLE_SYSTRAY_GTK_BACKEND) || defined(ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND)
+GdkPixbuf* SystemTrayIcon::convertQIconToPixbuf(const QIcon &icon)
+{
+    void (*callbackFreeImage)(guchar*, gpointer) =
+        [](guchar* image_bytes, gpointer)
+    {
+        delete[] image_bytes;
```

Comment:
Oh cool, real closure :)

Though… how does C++ know when the callback is gone for good?  C++ garbage collector magic even if it's passed to a C library?